src/projectscene/view/timeline/timelinecontext.cpp

```diff
@@ -11,14 +11,41 @@
 
 static constexpr double ZOOM_MIN = 0.001;
 static constexpr double ZOOM_MAX = 6000000.0;
+
+// on wheel event factor
 static constexpr int PIXELSSTEPSFACTOR = 5;
 
+// horizontal auto-scroll setup
+static constexpr double SCROLL_MARGIN_PX = 40.0;
+static constexpr double SCROLL_MIN_SPEED = 0.00001;
+static constexpr double SCROLL_MAX_SPEED = 0.025;
+
 using namespace au::projectscene;
 
+namespace {
+// calculate auto-scroll speed based on where the mouse cursor is
+// it's non linear range mapping:
+// from (0px, SCROLL_MARGIN_PX) to (SCROLL_MIN_SPEED, SCROLL_MAX_SPEED)
+double calculateScrollSpeed(double value, double inMin, double inMax, double outMin, double outMax) {
+    double sign = (muse::RealIsEqualOrMore(value, 0.0)) ? 1.0 : -1.0;
+    double absValue = std::abs(value);
+    double normalized = std::clamp((absValue - inMin) / (inMax - inMin), 0.0, 1.0);
+
+    double scaled = std::pow(normalized, 4);
+    double result = scaled * (outMax - outMin) + outMin;
+
+    return sign * result;
+}
+}
+
 TimelineContext::TimelineContext(QObject* parent)
     : QObject(parent)
 {
     m_snapTimeFormatter = std::make_shared<SnapTimeFormatter>();
+
+    m_scrollTimer = new QTimer();
+    m_scrollTimer->setInterval(16); // scroll at ~60 FPS
+    connect(m_scrollTimer, &QTimer::timeout, [this](){ autoScrollView(m_autoScrollStep); });
```

Comment:
I think it's better to use the timer by value (not as a pointer), or you need to add its deletion and initialization to null.